### ISSUE TITLE
Infinity login loop

### ISSUE THREAD

**Issue by destraty** [nextcloud/android#11575]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Opening app
2. Entering server's ip
3. Entering login and password
4. Confirming access

### Expected behaviour

Expected successfully login

### Actual behaviour

By the fact, we have infinity login loop. After confirming the access app throws me back to "enter server address" page. I've tried to reboot phone, reinstall app, install some early versions, clear cash and data but have no success. 

### Android version

11 RKQ1.201004.002

### Device brand and model

Xiaomi Redmi Note 8T

### Stock or custom OS?

Stock

### Nextcloud android app version

3.24.2

### Nextcloud server version
Armbian 23.02.2 Jammy with Linux 5.15.93-sunxi
nextcloud version: 26.0.1

### Using a reverse proxy?

I don't know

### Android logs

i can't login, so i can't view logs using "easy methods" 

### Server error logs

```bash
no error logs
```

**Comment by joshtrichards** [comment]
Does the Web UI work using the same URL?

**Comment by destraty** [comment]
> Does the Web UI work using the same URL?

Yeah, web ui works perfectly. 
O found some strange regularity.... This bug called "Grand Access loop" ando could be reproduced on Redmi 8x-series (like redmi 8, redmi 8T, redmi 8 pro). On other devices and models everything works perfectly.... (I've tried on iphone 14, redmi 7, redmi 10) 
I've found one issue with same problem: https://github.com/nextcloud/android/issues/9874.(happend on redmi 8, as not strange....) 
On every similar posts people recommends to add "overwritepeotocol => https" string, but this action has no result:(
Does anyone know something about this strange issues, which brakes my mind....

**Comment by destraty** [comment]
> > Does the Web UI work using the same URL?
> 
> Yeah, web ui works perfectly. O found some strange regularity.... This bug called "Grand Access loop" ando could be reproduced on Redmi 8x-series (like redmi 8, redmi 8T, redmi 8 pro). On other devices and models everything works perfectly.... (I've tried on iphone 14, redmi 7, redmi 10) I've found one issue with same problem: #9874 on redmi 8, as not strange....) On every similar posts people recommends to add "overwritepeotocol => https" string, but this action has no result:( Does anyone know something about this strange issues, which brakes my mind....

P.S. 

I checked this bug on several independent Redmi 8 devices and this error does appear. On each of the devices, either endless authorization occurs, or an "Access Forbiden Invalid requests" error appears.
I also tried doing a clean install of Linux and next cloud. The error hasn't gone away.

**Comment by destraty** [comment]
Bruh... Bad news, guys
Nextcloud app doesn't work at all on Android 11
For now, I can't find the method to fix it without flashing phone to Android 10 or Android 12
I've tried to downgrade my Android from 11 to 10 and omg, nextcloud works properly here(on Android 10 & miui 12) 
But when I upgraded back to Android 11 nextcloud broke again... 
Dear devs, @joshtrichards, pls pay attention on this issue, It's really annoying

**Comment by destraty** [comment]
Thanks all for fix it!

**Comment by bastianhjaeger** [comment]
Seeing the same behavior with android 14. 
Is it still an issue or does it just mimic like that?

**Comment by joshtrichards** [comment]
@bastianhjaeger Presumably the latter. In any case, please create a new Issue with the details of your situation. Log would be helpful (see repository README and/or get by installing the Dev version of the client via F-Droid in parallel).

**Comment by bastianhjaeger** [comment]
Yes. My issue was resolved by setting the protocol overwrite to https. Thx.
